homeassistant/components/pooldose/sensor.py

```diff
@@ -0,0 +1,208 @@
+        key="ph_calibration_slope",
+        translation_key="ph_calibration_slope",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.VOLTAGE,
+        suggested_display_precision=2,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfElectricPotential.MILLIVOLT,
+    ),
+    PooldoseSensorEntityDescription(
+        key="orp_calibration_type",
+        translation_key="orp_calibration_type",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        device_class=SensorDeviceClass.ENUM,
+        options=["off", "reference", "1_point"],
+    ),
+    PooldoseSensorEntityDescription(
+        key="orp_calibration_offset",
+        translation_key="orp_calibration_offset",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.VOLTAGE,
+        suggested_display_precision=2,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfElectricPotential.MILLIVOLT,
+    ),
+    PooldoseSensorEntityDescription(
+        key="orp_calibration_slope",
+        translation_key="orp_calibration_slope",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.VOLTAGE,
+        suggested_display_precision=2,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfElectricPotential.MILLIVOLT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: PooldoseConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up PoolDose sensor entities from a config entry."""
+    if TYPE_CHECKING:
+        assert config_entry.unique_id is not None
+
+    coordinator = config_entry.runtime_data.coordinator
+    client = config_entry.runtime_data.client
+    serial_number = config_entry.unique_id
+    device_properties = config_entry.runtime_data.device_properties
+
+    available = client.available_sensors()
+
+    async_add_entities(
+        PooldoseSensor(
+            coordinator,
+            description,
+            serial_number,
+            device_properties,
+        )
+        for description in SENSOR_DESCRIPTIONS
+        if description.key in available
+    )
+
+
+class PooldoseSensor(PooldoseEntity, SensorEntity):
+    """Sensor entity for the Seko PoolDose Python API."""
+
+    entity_description: PooldoseSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PooldoseCoordinator,
+        description: PooldoseSensorEntityDescription,
+        serial_number: str,
+        device_properties: dict[str, Any],
+    ) -> None:
+        """Initialize a PoolDose sensor entity."""
+        super().__init__(
+            coordinator,
+            serial_number,
+            device_info(device_properties, serial_number),
+            entity_description=description,
+        )
```

Comment:
if you move the `device_info` call to the `entity.py` and just put the `device_properties` to the base class, you could even remove the whole constructor as now you only have a constructor to reorder the parameters